src/multi.rs

```diff
@@ -214,25 +242,45 @@ impl MultiState {
             return Ok(());
         }
 
-        // Reap all consecutive 'zombie' progress bars from head of the list
+        // Assumption: if extra_lines is not None, then it has at least one line
+        debug_assert_eq!(
+            extra_lines.is_some(),
+            extra_lines.as_ref().map(Vec::len).unwrap_or_default() > 0
+        );
+
+        let mut reap_indices = vec![];
+
+        // Reap all consecutive 'zombie' progress bars from head of the list.
         let mut adjust = 0;
-        while let Some(index) = self.ordering.first().copied() {
-            let member = &self.members[index];
+        for index in self.ordering.iter() {
```

Comment:
You can use `&index` in the for-loop header to avoid having to deref for all uses.